substrate/cli/src/lib.rs

```diff
@@ -164,14 +163,17 @@ where
 {
 	panic_hook::set();
 
+	let full_version = service::Configuration::<<F>::Configuration, <F>::Genesis>
```

Comment:
this type would not be necessary to specify if this were a free function. i.e. it would just be 
`let full_version = service::full_version_from_strs(...)`

Reply:
Followed up in #496